apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/TimestampBound.cs

```diff
@@ -74,6 +74,12 @@ private TimestampBound(TimestampBoundMode mode, DateTime timestamp)
         /// </summary>
         public DateTime Timestamp { get; }
 
+        /// <summary>
+        /// If true, the Cloud Spanner-selected read timestamp is included in the
+        /// <see cref="Transaction" /> message that describes the transaction.
+        /// </summary>
+        public bool ReturnReadTimestamp { get; set; }
+
         /// <summary>
         /// Read at a timestamp where all previously committed transactions
         /// are visible.
```

Comment:
`Strong` is a singleton so if we set the `ReturnReadTimestamp` property once, then all strong reads will return this value. But changing it to `Strong()` will be a breaking change. What's the best way to fix this?

Reply:
TimestampBound is intended to be immutable. We definitely don't want to introduce a setter here.

Instead, I would suggest adding a `WithReturnReadTimestamp(bool)` method that returns a new instance with the given value. (I'd also suggest probably refactoring to have a single constructor that accepts all the property values.)

Reply:
Good idea. Done, including the refactoring.

The downside of the refactoring is that you can't have non-static default values (for TimeSpan and DateTime) so I had to pass these in explicitly. A bit tedious but not a huge issue given that it's all private.